### PR TITLE
Connect to DB: add firewall rule to access the db automatically

### LINKED ISSUE #114: Connect to DB: add firewall rule to access the db automatically
In order to connect to the database via Rider Azure plugin, we need to go on Azure portal to enable access to the database through firewall rules. We could do it automatically and update firewall rules when we trying to connect. We should prompt a user if he wants to do that and set the firewall access for a local IP when confirmed.

### PR DESCRIPTION
**Pull request by maartenba**
Fixes #114

fw-myip

In terms of the API's for getting public IP used: (checked on 2018-11-26)

* https://whatismyipaddress.com/api - "You may programmatically query that server"
* http://ip4.seeip.org - "You can use it without any real limit."
* http://v4.ident.me - No license info (http://api.ident.me/)

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/com/microsoft/intellij/serviceexplorer/azure/database/actions/SqlDatabaseConnectDataSourceAction.kt [hunk 1]
```diff
@@ -27,9 +27,8 @@ import com.microsoft.tooling.msservices.helpers.Name
 import com.microsoft.tooling.msservices.serviceexplorer.azure.database.sqldatabase.SqlDatabaseNode
 import com.microsoft.tooling.msservices.serviceexplorer.azure.database.sqlserver.SqlServerNode
 
-
 @Name("Connect to database")
-class ConnectDatabaseAction(private val databaseNode: SqlDatabaseNode)
+class SqlDatabaseConnectDataSourceAction(private val databaseNode: SqlDatabaseNode)
```
Comment (maartenba) [inline PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/com/microsoft/intellij/serviceexplorer/azure/database/actions/SqlDatabaseConnectDataSourceAction.kt:31]: This rename is not really part of functionality in this PR, but made sense to unify with other action names.

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/com/microsoft/intellij/serviceexplorer/azure/database/actions/SqlServerConnectDataSourceAction.kt [hunk 2]
```diff
@@ -27,7 +27,7 @@ import com.microsoft.tooling.msservices.helpers.Name
 import com.microsoft.tooling.msservices.serviceexplorer.azure.database.sqlserver.SqlServerNode
 
 @Name("Connect to server")
-class ConnectServerAction(private val databaseServerNode: SqlServerNode)
+class SqlServerConnectDataSourceAction(private val databaseServerNode: SqlServerNode)
```
Comment (maartenba) [inline PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/com/microsoft/intellij/serviceexplorer/azure/database/actions/SqlServerConnectDataSourceAction.kt:30]: This rename is not really part of functionality in this PR, but made sense to unify with other action names.

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/com/microsoft/intellij/serviceexplorer/azure/database/actions/AddCurrentIpAddressToFirewallAction.kt [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+                }
+            }
+        })
+    }
+
+    private fun requestAddFirewallRule(project: Project, sqlServer: SqlServer, publicIpAddressResult: PublicIpAddressProvider.Result) {
+        val ipAddressInput = Messages.showInputDialog(project,
+                "Add firewall rule for IP address:",
+                "Add firewall rule",
+                IconLoader.getIcon("/icons/Database.svg"),
```
Comment (sdubov) [inline PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/com/microsoft/intellij/serviceexplorer/azure/database/actions/AddCurrentIpAddressToFirewallAction.kt:89]: Firewall rules are set on Azure SQL Server level. Maybe show a SqlServer.svg icon? I've asked Konstantin to create a new icon for this dialog - https://youtrack.jetbrains.com/issue/RIDER-22115 (We can update it later)

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/com/microsoft/intellij/serviceexplorer/RiderNodeActionsMap.kt [hunk 4]
```diff
@@ -40,12 +40,14 @@ class RiderNodeActionsMap : NodeActionsMap() {
...
                     .add(SqlServerOpenInBrowserAction::class.java)
-                    .add(ConnectServerAction::class.java)
+                    .add(SqlServerAddCurrentIpAddressToFirewallAction::class.java)
+                    .add(SqlServerConnectDataSourceAction::class.java)
                     .build()
 
             node2Actions[SqlDatabaseNode::class.java] = ImmutableList.Builder<Class<out NodeActionListener>>()
                     .add(SqlDatabaseOpenInBrowserAction::class.java)
-                    .add(ConnectDatabaseAction::class.java)
+                    .add(SqlDatabaseAddCurrentIpAddressToFirewallAction::class.java)
```
Comment (sdubov) [inline PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/com/microsoft/intellij/serviceexplorer/RiderNodeActionsMap.kt:49]: Do we want to add this action on DB level as well?

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/com/microsoft/intellij/serviceexplorer/azure/database/actions/AddCurrentIpAddressToFirewallAction.kt (hunk 3 above)
Comment (maartenba) [reply PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/com/microsoft/intellij/serviceexplorer/azure/database/actions/AddCurrentIpAddressToFirewallAction.kt:89]: Sounds good, we can replace it later.

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/com/microsoft/intellij/serviceexplorer/RiderNodeActionsMap.kt (hunk 4 above)
Comment (maartenba) [reply PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/com/microsoft/intellij/serviceexplorer/RiderNodeActionsMap.kt:49]: Feels easy? Usually you will want to connect to a specific database, and then get the firewall configured. Should we remove it here or not?

Comment (sdubov) [reply PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/com/microsoft/intellij/serviceexplorer/RiderNodeActionsMap.kt:49]: Agreed to have it for both nodes.

**Comment by maartenba** [comment]
FYI @sdubov - rebased on `191`

**Pull request merged**